### PR TITLE
Set comma expression deprecation release and implicit string concatenation

### PR DESCRIPTION
**Pull request by mathias-lang-sociomantic**
Please do not merge until dlang/dmd#6155 is merged.

### CODE REVIEW

**Comment by mathias-lang-sociomantic** [comment]
The DMD P.R. was merged :tada:

**Comment by mathias-lang-sociomantic** [comment]
Pingity ping

**Comment by AndrejMitrovic** [comment]
Is the build failure unrelated to the PR, can we merge regardless?

File: deprecate.dd [hunk 1]
```diff
@@ -57,6 +58,36 @@ $(SPEC_S Deprecated Features,
...
+$(H4 Rationale)
+    $(P This is a very early feature of the language, which is nowadays totally
+        covered by the concatenation operator: it is performed at compile time
+        for constants and doesn't result in memory allocation.)
+
+    $(P However, having implicit concatenation can and did result in hard
+        to spot bugs, for example:)
+
+        ---
+        string[] arr = [ "Hello", "buggy" "World" ];
```
Comment (AndrejMitrovic) [inline deprecate.dd]: I'd use a more realistic example. The classic mistake is made when the list of strings is separated on each line:

```D
string[] names =
[
    "Anna",
    "Michael"
    "Emma",
    "David"
];
```

**Comment by AndrejMitrovic** [comment]
> Is the build failure unrelated to the PR, can we merge regardless?

Ah all pulls are failing, nevermind me.

**Comment by mathias-lang-sociomantic** [comment]
Updated, sorry for the delay

File: deprecate.dd [hunk 2]
```diff
@@ -57,6 +58,42 @@ $(SPEC_S Deprecated Features,
...
 
 
+$(H3 $(DEPNAME Implicit string concatenation))
+    $(P Currently, two adjacent strings are implicitly concatenated:
+
+        ---
+        string foo = "Hello" "World";
+        ---
+
+        This feature is handy for long string that spans multiple lines,
```
Comment (AndrejMitrovic) [inline deprecate.dd]: `for a long`

**Comment by AndrejMitrovic** [comment]
Can you fix the missing `a` and I'll merge?

**Comment by mathias-lang-sociomantic** [comment]
Fixed

**Pull request merged**